### PR TITLE
Adds a ForeignTableEntity and Subtype

### PR DESCRIPTION
**Pull request by RussellSpitzer**

### CODE REVIEW

File: polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ */
+package org.apache.polaris.core.entity;
+
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.rest.RESTUtil;
+
+public class ForeignTableEntity extends TableLikeEntity {
+
+  public static final String FOREIGN_SOURCE_KEY = "_source";
```
Comment (eric-maynard) [inline polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java:27]: Is this the format?

Comment (RussellSpitzer) [reply polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java:27]: Yeah sorry, I'm stuck in very old Spark where this was called source ... I think

File: polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  public ForeignTableEntity(PolarisBaseEntity sourceEntity) {
+    super(sourceEntity);
+  }
+
+  public String getSource() {
+    return getPropertiesAsMap().get(FOREIGN_SOURCE_KEY);
```
Comment (eric-maynard) [inline polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java:34]: I worry that we need to somehow prevent conflicts with other properties... maybe we can skip that for now.

Comment (RussellSpitzer) [reply polaris-core/src/main/java/org/apache/polaris/core/entity/ForeignTableEntity.java:39]: yeah that's why I dropped the _ in here for now. I agree will need to do something safer later

**Review by eric-maynard — APPROVED** [review]

**Pull request merged**